docs/dev/spark-devel-env.rst

```diff
@@ -0,0 +1,110 @@
+Set up ListenBrainz Spark development environment
+=================================================
+
+There are two distinct part of the ListenBrainz development environment:
+
+1. listenbrainz -- the actual webserver components of ListenBrainz
+2. listenbrainz_spark -- the spark environment used for features that involve data processing (stats, recommendations etc.)
+
+If you're just working on adding a feature to the ListenBrainz webserver, you **do not** need
+to set up the Spark development environment. However, if you're looking to add
+a new stat or improve our fledgling recommender system, you'll need both the webserver
+and the spark development environment.
+
+This guide should explain how to develop and test new features for ListenBrainz that use Spark. 
+
+Set up the webserver
+--------------------
+The spark environment is dependent on the webserver. Follow the steps in the :doc:`guide to set up the webserver environment <devel-env>`. 
+
+Create listenbrainz_spark/config.py
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The spark environment needs a config.py in the listenbrainz_spark/ dir. Create it by copying from the sample config file.
+
+.. code-block:: bash
+ 
+    cp listenbrainz_spark/config.py.sample listenbrainz_spark/config.py
+
+
+Initialize ListenBrainz Spark containers
+----------------------------------------
+
+Run the following command to build the spark containers.
+
+.. code-block:: bash
+
+    ./spark_develop.sh build
+    
+The first time you build the containers, you also need to format the ``namenode`` 
+container. 
+
+.. code-block:: bash
+
+    ./spark_develop.sh format
+
+Your development environment is now ready. Now, let's actually see ListenBrainz Spark
+in action!
+
+
+Bring containers up
+--------------------
+
+Start the ListenBrainz Spark containers by executing ``spark_develop.sh up``. 
+
+.. code-block:: bash
+
+    ./spark_develop.sh up
+
+Import data into the spark environment
+--------------------------------------
+
+We provide small data dumps that are helpful for working with real ListenBrainz data.
+Download and import a data dump into your spark environment using the following 
+commands.
+
+.. code-block:: bash
+
+    ./spark_develop.sh run playground python spark_manage.py upload_listens -i
```

Comment:
It should be `./spark_develop.sh run request_consumer python spark_manage.py upload_listens -i`

Reply:
Nice catch! Thanks!